test/unit/org/apache/cassandra/db/compaction/LeveledCompactionStrategyTest.java

```diff
@@ -367,4 +379,120 @@ public void testNewRepairedSSTable() throws Exception
         assertTrue(unrepaired.manifest.getLevel(1).contains(sstable2));
         assertFalse(repaired.manifest.getLevel(1).contains(sstable2));
     }
+
+    @Test
+    public void testReduceScopeL0L1() throws IOException
```

Comment:
nit: It might also be nice to have a quick test that verifies `reduceScopeForLimitedSpace()` aborts when you throw an L2 SSTable in there.